components/server/src/auth/auth-provider-service.ts

```diff
@@ -28,11 +28,18 @@ export class AuthProviderService {
      */
     async getAllAuthProviders(): Promise<AuthProviderParams[]> {
         const all = await this.authProviderDB.findAll();
-        return all.map(this.toAuthProviderParams.bind(this));
+        const transformed = all.map(this.toAuthProviderParams.bind(this));
+
+        // as a precaution, let's remove duplicates
+        const unique = transformed.reduce((prev, current) => {
```

Comment:
How do we end up with duplicates? Also just filtering any of the dupes out seem like potentially resulting in strange surprising behavior. Shouldn't we at least log errors in such cases?

Reply:
> How do we end up with duplicates? 

potentially, because of different casing. :-(



> Shouldn't we at least log errors in such cases?

yes

Reply:
will add log statement in such cases

Reply:
done